client/components/FirstVendor.js

```diff
@@ -26,34 +36,54 @@ class FirstVendor extends React.Component {
       promptAdjustPosition: { x: 0, y: 2, z: 0 },
       promptIndex: 0,
       responseAdjustPosition: { x: 0, y: 1, z: 0 },
-      grade: '',
+      grade: {response: '', attempts: 3},
       language: {
         langCode: 'es-419',
         fromLang: 'es',
         toLang: 'en'
-      }
+      },
+      countdown: {counting: false, timer: 3},
+      repeat: false,
+      gameOver: false
     }
     this.handleVendorClick = this.handleVendorClick.bind(this)
+    this.listenToUser = this.listenToUser.bind(this)
   }
 
   handleVendorClick() {
-    if (Object.keys(this.props.currentPrompt).length) {
-      this.props.setCurrentPrompt(this.props.prompts[this.state.promptIndex])
-    } else {
-      this.props.setCurrentPrompt(this.props.prompts[0])
-    }
-    let index = this.state.promptIndex
-    if (index < this.props.prompts.length - 1) {
-      index++
-      this.setState({ promptIndex: index })
-    } else {
-      this.setState({ promptIndex: 0 })
+    if(!this.state.gameOver) {
+      if(!this.state.repeat) {
+        if (Object.keys(this.props.currentPrompt).length) {
+          this.props.setCurrentPrompt(this.props.prompts[this.state.promptIndex])
+        } else {
+          this.props.setCurrentPrompt(this.props.prompts[0])
+        }
+        let index = this.state.promptIndex
+        if (index < this.props.prompts.length - 1) {
```

Comment:
Why it is index < this.props.prompts.length -1  and not this.props.prompts.length